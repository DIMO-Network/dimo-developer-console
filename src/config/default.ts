export const appName = 'Developer Console';

export const LOGIN_PAGES = ['/sign-in', '/sign-up', '/email-recovery'];

// Just a placeholder for redirection purposes
export const VALIDATION_PAGES = ['/valid-tzd', '/verify-email'];

export const API_PATH = '/api';

export const UNPROTECTED_PATHS = ['/api/auth'];

export const ROLES = ['Collaborator'];

export const REGIONS = [
  'North America',
  'Latin America & Caribbean',
  'Oceania',
  'Northern & Western Europe',
  'Africa',
  'East & Southeast Asia',
  'Middle East & Central Asia ',
  'South Asia',
  'Eastern Europe',
];

export const DEVELOPER_TYPES = [
  'OEM',
  'Insurance',
  'Maintenance',
  'Fleet Management',
  'Vehicle Sales',
  'Utilities',
  'Financing',
  'EV',
  'DePIN',
  'Data',
  'Others',
];

export const backendUrl = 'http://localhost:3001/';

export const frontendUrl = 'http://localhost:3000/';

export const RAINBOW_PROJECT = {
  ID: '528803928611a7781fb6b23eaf232224',
  NAME: 'Dimo Developer Console',
};

// 80002 - POLYGON AMOY
export const CONTRACT_NETWORK = BigInt(80_002);

// DIMO TOKEN AMOY
export const DC_ADDRESS: `0x${string}` = '0x21cFE003997fB7c2B3cfe5cf71e7833B7B2eCe10';

// DIMO CREDIT AMOY
export const DCX_ADDRESS: `0x${string}` = '0x49c120f4C3c6679Ebd357F2d749E4D1C03598d65';

// DIMO LICENSE AMOY
export const DLC_ADDRESS: `0x${string}` = '0xdb6c0dBbaf48b9D9Bcf5ca3C45cFF3811D70eD96';

// WRAPPED MATIC AMOY
export const WMATIC: `0x${string}` = '0x360ad4f9a9A8EFe9A8DCB5f461c4Cc1047E1Dcf9';

// SwapRouter AMOY
export const SwapRouterAddress: `0x${string}` =
  '0x52D03752872F30B6AcD26979FD5EB7213bA62DaF';

export const masFeePerGas = 60000000000;
export const gasPrice = 43000000000;

export const MINIMUM_CREDITS = 10000;

export const DIMO_CONTRACT_ADDRESS = '0x523d4a08cf149f1Ada8113B3b3400234236Bb5E8';

export const ISSUED_TOPIC: `0x${string}` =
  '0x7533f62ec6601bf9c87f8d96bf756b4b495e2a0e26ec9284e4927926ed6b3afd';
