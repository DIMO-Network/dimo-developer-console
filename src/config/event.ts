export const otpLoginProcessed = 'otp-login-processed';
