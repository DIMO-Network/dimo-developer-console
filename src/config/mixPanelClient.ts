import mixpanel from 'mixpanel-browser';

export const initMixPanel = () => {
  const mixPanelToken = process.env.NEXT_PUBLIC_MIXPANEL_TOKEN;
  const vercelEnv = process.env.VERCEL_ENV;

  if (!mixPanelToken!) {
    console.error('Mixpanel token is not defined');
    return;
  }

  mixpanel.init(mixPanelToken!, {
    persistence: 'localStorage',
    debug: vercelEnv === 'development',
    ignore_dnt: true,
    autocapture: true,
  });
};
