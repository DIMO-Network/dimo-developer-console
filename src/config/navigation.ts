import {
  HomeIcon,
  IntegrationIcon,
  MonitorHeartIcon,
  SettingsIcon,
  SummarizeIcon,
  SupportAgentIcon,
} from '@/components/Icons';

const APP_DETAILS_REGEX = /^\/app\/details\/[^/]+$/;
const LICENSE_DETAILS_REGEX = /^\/license\/details\/[^/]+$/;
const LICENSED_VEHICLES_REGEX = /^\/license\/vehicles\/[^/]+$/;
export const getPageTitle = (path: string) => {
  const staticPageTitle = pageTitles[path];
  if (staticPageTitle) return staticPageTitle;
  if (APP_DETAILS_REGEX.test(path)) return 'App Details';
  if (LICENSE_DETAILS_REGEX.test(path)) return 'License Details';
  if (LICENSED_VEHICLES_REGEX.test(path)) return 'Licensed Vehicles';
};

const pageTitles: Record<string, string> = {
  '/': 'Home',
  '/app': 'Home',
  '/integrations': 'Integrations',
  '/api-status': 'API Status',
  '/settings': 'Settings',
};

export const mainMenu = [
  {
    label: 'Home',
    icon: HomeIcon,
    iconClassName: 'h-5 w-5',
    link: '/app',
    external: false,
    disabled: false,
  },
  {
    label: 'Integrations',
    icon: IntegrationIcon,
    iconClassName: 'h-5 w-5 fill-white stroke-white stroke-1',
    link: '/integrations',
    external: false,
    disabled: false,
    subMenu: [
      {
        label: 'Webhooks',
        link: '/integrations/webhooks',
        external: false,
      },
    ],
  },
  {
    label: 'Support',
    icon: SupportAgentIcon,
    iconClassName: 'h-5 w-5',
    link: 'https://discord.com/channels/892438668453740634/940719111971946546',
    external: true,
    disabled: false,
  },
  {
    label: 'Documentation',
    icon: SummarizeIcon,
    iconClassName: 'h-5 w-5',
    link: 'https://docs.dimo.zone/developer-platform',
    external: true,
    disabled: false,
  },
  {
    label: 'API Status',
    icon: MonitorHeartIcon,
    iconClassName: 'h-5 w-5',
    link: '/api-status',
    external: false,
    disabled: true,
  },
];

export const bottomMenu = [
  {
    label: 'Settings',
    icon: SettingsIcon,
    iconClassName: 'h-5 w-5 fill-grey-200',
    link: '/settings',
    external: false,
    disabled: false,
  },
];
