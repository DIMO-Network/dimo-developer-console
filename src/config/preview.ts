export const backendUrl = 'https://console-staging-api.dimo.org/';

export const frontendUrl = 'https://console-staging.dimo.org/';
