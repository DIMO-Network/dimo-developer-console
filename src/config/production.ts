export const backendUrl = 'https://console-api.dimo.xyz/';

export const frontendUrl = 'https://console.dimo.xyz/';

// 137 - MAINNET
export const CONTRACT_NETWORK = BigInt(137);

// DIMO TOKEN MAINNET
export const DC_ADDRESS = '0xE261D618a959aFfFd53168Cd07D12E37B26761db';

// DIMO LICENSE MAINNET
export const DLC_ADDRESS = '0x9A9D2E717bB005B240094ba761Ff074d392C7C85';