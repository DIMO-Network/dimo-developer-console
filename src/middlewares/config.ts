import { AuthMiddleware } from '@/middlewares/auth.middleware';

export const middlewares = [AuthMiddleware];
