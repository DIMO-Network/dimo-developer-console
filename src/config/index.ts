import * as defaultConfig from './default';
import * as productionConfig from './production';
import * as previewConfig from './preview';

type Configuration = {
  appName: string;
  LOGIN_PAGES: string[];
  VALIDATION_PAGES: string[];
  API_PATH: string;
  UNPROTECTED_PATHS: string[];
  ROLES: string[];
  REGIONS: string[];
  DEVELOPER_TYPES: string[];
  backendUrl: string;
  frontendUrl: string;
  RAINBOW_PROJECT: Record<string, string>;
  CONTRACT_NETWORK: bigint;
  DLC_ADDRESS: `0x${string}`;
  DC_ADDRESS: `0x${string}`;
  DCX_ADDRESS: `0x${string}`;
  WMATIC: `0x${string}`;
  SwapRouterAddress: `0x${string}`;
  masFeePerGas: number;
  gasPrice: number;
  desiredAmountOfDCX: number;
  desiredAmountOfDimo: number;
};

export const getConfig = (): Configuration => {
  // Determine the current environment
  const env = process.env.VERCEL_ENV;

  // Select the appropriate configuration to merge with default based on the environment
  let environmentConfig = {};
  switch (env) {
    case 'production':
      environmentConfig = productionConfig;
      break;
    case 'preview':
      environmentConfig = previewConfig;
      break;
    // Assuming 'development' should fall back to default
    case 'development':
    default:
      environmentConfig = {};
      break;
  }

  // Use lodash to deeply merge the default configuration with the environment-specific configuration
  return {
    ...defaultConfig,
    ...environmentConfig,
  } as Configuration;
};

export default getConfig();
