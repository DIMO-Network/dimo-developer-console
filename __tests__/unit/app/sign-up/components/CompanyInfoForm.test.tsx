import axios from 'axios';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { useSearchParams } from 'next/navigation';

import SignUpPage from '@/app/sign-up/page';
import { regions } from '@/config/default';
import nock from 'nock';

jest.mock('next/navigation');

const replace = window.location.replace;

axios.defaults.adapter = 'http';

beforeEach(() => {
  Object.defineProperty(window, 'location', {
    value: { replace: jest.fn() },
  });
});

afterEach(() => {
  window.location.replace = replace;
});

describe('SignUpPage', () => {
  it('renders the sign in page at the company info form', () => {
    jest.mocked<any>(useSearchParams).mockImplementationOnce(() => ({
      get: jest.fn(() => 'company-information'),
    }));

    const { container } = render(<SignUpPage />);

    expect(container).toMatchSnapshot();
  });

  it('renders wrong url message at build for form', async () => {
    jest.mocked<any>(useSearchParams).mockImplementationOnce(() => ({
      get: jest.fn(() => 'company-information'),
    }));

    render(<SignUpPage />);

    const inputBuildFor = screen.getByRole(
      'company-website-input'
    ) as HTMLInputElement;
    const finishBtn = screen.getByRole('finish-button');
    fireEvent.change(inputBuildFor, { target: { value: 'test' } });
    fireEvent.click(finishBtn);

    const messageElm = await screen.findByText(
      'This field must be a valid URL'
    );

    expect(messageElm).toBeInTheDocument();
  });

  it('submitting the build for form', async () => {
    jest.mocked<any>(useSearchParams).mockImplementationOnce(() => ({
      get: jest.fn(() => 'company-information'),
    }));

    nock('http://localhost:3000')
      .persist()
      .intercept('/api/user', 'OPTIONS')
      .reply(200)
      .put('/api/user')
      .reply(200);

    window.location.replace = jest.fn();

    render(<SignUpPage />);

    const regionContainer = screen.getByRole(
      'company-region'
    ) as HTMLInputElement;
    fireEvent.click(regionContainer);

    const [, optionElm] = await screen.findAllByText(regions[0]);
    fireEvent.click(optionElm);

    const finishBtn = screen.getByRole('finish-button');
    fireEvent.click(finishBtn);

    await waitFor(() =>
      expect(window.location.replace).toHaveBeenCalledWith('/sign-up')
    );
  });
});
