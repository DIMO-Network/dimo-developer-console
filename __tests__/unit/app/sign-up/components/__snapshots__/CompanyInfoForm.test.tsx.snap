// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage renders the sign in page at the company info form 1`] = `
<div>
  <main
    class="sign-up"
  >
    <div
      class="sign-up__content"
    >
      <article
        class="sign-up__form"
      >
        <section
          class="sign-up__header"
        >
          <img
            alt="DIMO Logo"
            class="w-44 h-6"
            src="/images/build-on-dimo.png"
          />
          <p>
            Our last question
          </p>
        </section>
        <form
          class="flex flex-col gap-4 w-full max-w-sm pt-4"
        >
          <label
            class="label text-xs text-medium"
            for="name"
          >
            Company name
            <input
              class="text-field"
              name="name"
              placeholder="DIMO"
              role="company-name-input"
              type="text"
            />
          </label>
          <label
            class="label text-xs text-medium"
            for="website"
          >
            Company website
            <input
              class="text-field"
              name="website"
              placeholder="www.dimo.zone"
              role="company-website-input"
              type="text"
            />
          </label>
          <label
            class="label text-xs text-medium"
            for="region"
          >
            Main Operating Region *
            <div
              class="select-field"
              role="company-region"
            >
              <select
                name="region"
                placeholder="Select"
                role="company-region-select"
              >
                <option />
                <option
                  value="North America"
                >
                  North America
                </option>
                <option
                  value="Latin America & Caribbean"
                >
                  Latin America & Caribbean
                </option>
                <option
                  value="Oceania"
                >
                  Oceania
                </option>
                <option
                  value="Northern & Western Europe"
                >
                  Northern & Western Europe
                </option>
                <option
                  value="Africa"
                >
                  Africa
                </option>
                <option
                  value="East & Southeast Asia"
                >
                  East & Southeast Asia
                </option>
                <option
                  value="Middle East & Central Asia "
                >
                  Middle East & Central Asia 
                </option>
                <option
                  value="South Asia"
                >
                  South Asia
                </option>
                <option
                  value="Eastern Europe"
                >
                  Eastern Europe
                </option>
              </select>
              <p
                class=""
                role="company-region-text"
              >
                Select
              </p>
              <svg
                class="w-4 h-4"
                fill="none"
                role="chevron-down-icon"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m19.5 8.25-7.5 7.5-7.5-7.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              <div
                class="custom-menu"
              >
                <div
                  class="custom-item"
                >
                  North America
                </div>
                <div
                  class="custom-item"
                >
                  Latin America & Caribbean
                </div>
                <div
                  class="custom-item"
                >
                  Oceania
                </div>
                <div
                  class="custom-item"
                >
                  Northern & Western Europe
                </div>
                <div
                  class="custom-item"
                >
                  Africa
                </div>
                <div
                  class="custom-item"
                >
                  East & Southeast Asia
                </div>
                <div
                  class="custom-item"
                >
                  Middle East & Central Asia 
                </div>
                <div
                  class="custom-item"
                >
                  South Asia
                </div>
                <div
                  class="custom-item"
                >
                  Eastern Europe
                </div>
              </div>
            </div>
          </label>
          <div
            class="flex flex-col pt-4"
          >
            <button
              class="button primary"
              role="finish-button"
              type="submit"
            >
              Finish sign up
            </button>
          </div>
        </form>
      </article>
    </div>
  </main>
</div>
`;
